### ISSUE TITLE
NettyChannelHandler中对链接异常的处理

### ISSUE THREAD

**Issue by aqiangsoft** [weibocom/motan#783]
建议在NettyChannelHandler中channel exceptionCaught时发现有链接由于与motan请求头header定义的magic魔数不匹配时对异常进行细化处理  像这种不管是框架还是业务应用都无法处理的异常不应该往外部抛出

**Comment by aqiangsoft** [comment]
like this: NettyChannelHandler exceptionCaught: remote=/185.54.91.37:40034 local=/120.79.229.16:8011 event=com.weibo.api.motan.exception.MotanFrameworkException: error_message: NettyDecoder transport header not support, type: 18245, status: 503, error_code: 20001,r=null
io.netty.handler.codec.DecoderException: com.weibo.api.motan.exception.MotanFrameworkException: error_message: NettyDecoder transport header not support, type: 18245, status: 503, error_code: 20001,r=null

**Comment by rayzhang0603** [comment]
NettyChannelHandler中channel exceptionCaught时的行为是log异常日志、断开链接，这个是预期的行为。

除魔数异常外，遇到数据读取异常（长度错误）、数据超长等异常的期望行为都是记录异常、断开链接，这些都是统一由NettyChannelHandler的hannel exceptionCaught处理的

**Comment by aqiangsoft** [comment]
是的  确实只是日志记录行为  我想表达的是可否对这些异常作细化处理 避免框架error级别异常和应用业务error异常混合 比如是否可支持分文件记录  我们目前线上rpc层的NettyDecoder transport header not support, type异常经常和业务异常混在一起  比较不好排查问题 虽然此类异常不多  但对日志文件监控还是有一些影响

**Comment by rayzhang0603** [comment]
我觉得有两个办法解决这个问题：

1、排查问题时，**如果只关心业务异常**，可以`grep -v 'MotanFrameworkException'`去掉motan框架日志。

2、可以实现`LogService`接口，替换掉LoggerUtil中的默认实现，这样可以在log日志时对异常做定制化处理。

建议通过方式1解决。

**Comment by aqiangsoft** [comment]
@rayzhang0603 
 1.感谢提供的解决方案  第一种通过运维来避免很方便直接 原来系统是想对error日志作增量告警 目前看来 得使用关键链路异常在代码层面处理告警通知了
2. 第二种方案评估了一下LogService属于motan-core模块 影响层面较广 考虑到后续与motan官方保持持续升级和维护 还是选择第一种方式来解决日志排查的问题比较稳妥 毕竟不会影响业务链路
